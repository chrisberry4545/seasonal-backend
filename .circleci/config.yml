version: 2
jobs:
  test:
    docker:
      - image: circleci/node:8.10

    steps:
      - checkout

      - setup_remote_docker

      - run:
          name: Output dir
          command: echo $PWD
      - run:
          name: Run Docker
          command: docker-compose run app npm test

      - run:
          name: Run tests
          command: docker exec app npm run test

  deploy:
    docker:
      - image: circleci/node:8.10

    steps:
      - checkout

      - restore_cache:
          keys:
          - v1-dependencies-
          # fallback to using the latest cache if no exact match is found
          - v1-dependencies-

      - run:
          name: Copy env config
          command: |
            cp .env-ci.yml .env.yml

      - run:
          name: Install Serverless CLI and dependencies
          command: |
            sudo npm i -g serverless
            npm install

      - run:
          name: Run tests
          command: npm test

      - run:
          name: Build dist version
          command: |
            npm run build
            npm run move-dist-to-top-level

      - save_cache:
          paths:
            - node_modules
          key: v1-dependencies-

      - run:
          name: Deploy application
          command: sls deploy --stage prod

workflows:
  version: 2
  ci:
    jobs:
      - deploy:
          filters:
            branches:
              only:
                - master
      - test
