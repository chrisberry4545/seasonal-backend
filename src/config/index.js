module.exports = {
  AIRTABLE_API_KEY: process.env.AIRTABLE_API_KEY,
  SKIP_CACHE: process.env.SKIP_CACHE || false
};
