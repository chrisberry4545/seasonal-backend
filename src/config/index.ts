export const SKIP_CACHE = process.env.SKIP_CACHE || false;
export const SEASON_ENDPOINT =
  process.env.SEASON_ENDPOINT || 'season';
export const FOOD_ENDPOINT =
  process.env.FOOD_ENDPOINT || 'food';
export const SEASON_WITH_FOOD_ENDPOINT =
  process.env.SEASON_WITH_FOOD_ENDPOINT || 'season-with-food';
export const SEASON_WITH_RECIPES_ENDPOINT =
  process.env.SEASON_WITH_RECIPES_ENDPOINT || 'season-with-recipes';
export const COUNTRY_ENDPOINT =
  process.env.COUNTRY_ENDPOINT || 'country';
export const V2_ENDPOINT = process.env.V2_ENDPOINT || 'v2';
export const DB_CONNECTION_STRING = process.env.DB_CONNECTION_STRING;

export const DEFAULT_COUNTRY_ID =
  process.env.DEFAULT_COUNTRY_ID || '83533319-9b7a-4c79-9261-4c02bc6b4d0d';
