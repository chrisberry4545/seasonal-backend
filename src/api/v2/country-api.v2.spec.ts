import {
  app
} from '../../app';

import supertest, { Response } from 'supertest';
import { V2_ENDPOINT, COUNTRY_ENDPOINT } from '../../config';
import { ICountry, IRegion } from '@chrisb-dev/seasonal-shared';

const countryUrl = `${V2_ENDPOINT}/${COUNTRY_ENDPOINT}`;

describe('Get countries', () => {
  let response: Response;
  let result: ICountry[];
  let allRegions: IRegion[];
  beforeAll(async () => {
    response = await supertest(app).get(`/${countryUrl}`);
    result = response.body;
    allRegions = result.reduce((regions, country) => [
      ...regions,
      ...country.regions
    ], [] as IRegion[]);
  });

  test('Returns a status of 200', () => {
    expect(response.status).toBe(200);
  });
  test('Retrieves a single food item', async () => {
    expect(result).toMatchSnapshot();
  });
  test('Returns multiple countries', () => {
    expect(result.length > 0).toBe(true);
  });
  test('Populates every countries id', () => {
    const areAllIdsDefined =
      result.every((country) => country.id !== undefined);
    expect(areAllIdsDefined).toBe(true);
  });
  test('Populates every countries name', () => {
    const areAllNamesDefined =
      result.every((country) => country.name !== undefined);
    expect(areAllNamesDefined).toBe(true);
  });
  test('Every countries has regions', () => {
    const doAllCountriesHaveRegions =
      result.every((country) => country.regions.length > 0);
    expect(doAllCountriesHaveRegions).toBe(true);
  });
  test('Populates every regions name', () => {
    const allRegionsNamesDefined = allRegions
      .every((region) => region.name !== undefined);
    expect(allRegionsNamesDefined).toBe(true);
  });
  test('Populates every regions code', () => {
    const allRegionsCodesDefined = allRegions
      .every((region) => region.code !== undefined);
    expect(allRegionsCodesDefined).toBe(true);
  });
});
