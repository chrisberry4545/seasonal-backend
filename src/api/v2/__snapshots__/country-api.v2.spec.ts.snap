// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get countries Retrieves a single food item 1`] = `
Array [
  Object {
    "id": "1fc52423-eb83-4cd9-9fdd-b6f9cb323c37",
    "name": "UK",
    "regions": Array [
      Object {
        "code": "gbr",
        "latLng": Object {
          "lat": 50,
          "lng": 50,
        },
        "name": "UK",
      },
    ],
  },
  Object {
    "id": "d6e57673-eee8-444a-b7be-d9ab553052cf",
    "name": "Australia",
    "regions": Array [
      Object {
        "code": "aus-sydney",
        "latLng": Object {
          "lat": -50,
          "lng": -50,
        },
        "name": "Sydney",
      },
    ],
  },
]
`;
